### PR TITLE
Fix overflow in `searchsorted*`

### LINKED ISSUE #285: Overflow in `searchsorted*`
```julia
julia> soa = OffsetArray([2,2,3], typemax(Int)-4)
3-element OffsetArray(::Vector{Int64}, 9223372036854775804:9223372036854775806) with eltype Int64 with indices 9223372036854775804:9223372036854775806:
 2
 2
 3

julia> searchsortedfirst(soa, 2)
-3
```

Found this in the `Base` tests, we should have these tests added here.

### PR DESCRIPTION
**Pull request by jishnub**
Fixes #285, but this doesn't fix all issues (as certain tests are broken). However, I think `searchsortedfirst` and `searchsortedlast` should be fixed.

### CODE REVIEW

**Review by johnnychen94 — APPROVED** [review]

**Comment by johnnychen94** [comment]
Hmmm, the failure on 1.0 is concerning. Do you know why that happens?

**Comment by jishnub** [comment]
Yes, should be fixed now. I had made a mistake in the test, but since empty ranges are equal in 1.6+, the test had passed.

**Comment by jishnub** [comment]
I feel like we're unnecessarily complicating matters here. We should simply pass the `searchsorted` to the parent, and add the offset. I've tried this locally and this resolves all issues.

**Comment by jishnub** [comment]
I think the codecov changes are ok, looks good to merge

File: src/OffsetArrays.jl [hunk 1]
```diff
@@ -771,74 +771,25 @@ centered(A::AbstractArray, cp::Dims=center(A)) = OffsetArray(A, .-cp)
...
-        else
-            a = searchsortedfirst(v, x, max(lo,ilo), m, o)
-            b = searchsortedlast(v, x, m, min(hi,ihi), o)
-            return a : b
-        end
+# we may pass the searchsorted* functions to the parent, and wrap the offset
+for f in [:searchsortedfirst, :searchsortedlast, :searchsorted]
+    _safe_f = Symbol("_safe_" * String(f))
+    @eval function $_safe_f(v::OffsetArray, x, ilo, ihi, o::Base.Ordering)
+        offset = firstindex(v) - firstindex(parent(v))
```
Comment (johnnychen94) [inline src/OffsetArrays.jl:778]: What if `v` is an offset vector of another offset vector?

**Review by johnnychen94 — APPROVED** [review]

File: src/OffsetArrays.jl (hunk 1 above)
Comment (jishnub) [reply src/OffsetArrays.jl:778]: Maybe I'll add some tests, good to be sure

**Pull request merged**